### PR TITLE
Gif task table

### PR DESCRIPTION
**Pull request by jzwart**
Creates a gif task yaml for creating .png files based on `create_storm_frame` from #33 for including in a gif. 

I wasn't able to build this because I'm still hung up on the netcdf windows vs. mac error and `6_visualize.yml` is dependent on everything upstream.

### CODE REVIEW

**Comment by jzwart** [comment]
yup, in the office now and I'll get them

File: 6_visualize/src/create_gif_tasks.R [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  # function to sprintf a bunch of key-value (string-variableVector) pairs, then
+  # paste them together with a good separator for constructing remake recipes
+  psprintf <- function(..., sep='\n      ') {
+    args <- list(...)
+    strs <- mapply(function(string, variables) {
+      spargs <- if(string == '') list(variables) else c(list(string), as.list(variables))
+      do.call(sprintf, spargs)
+    }, string=names(args), variables=args)
+    paste(strs, collapse=sep)
+  }
```
Comment (aappling-usgs) [inline 6_visualize/src/create_gif_tasks.R]: are you liking this function? should we formalize it in `scipiper` one of these days?

File: 6_visualize.yml [hunk 2]
```diff
@@ -43,3 +46,27 @@ targets:
...
+  # out=folder to hold .ind and data files corresponding to shared cache or everybody's local build.
+  # log=folder for the few indicator files that don't correspond to a data file.
+  6_visualize/tmp:
+    command: dir.create(target_name, recursive=I(TRUE), showWarnings=I(FALSE))
+  6_visualize/log:
+    command: dir.create(target_name, recursive=I(TRUE), showWarnings=I(FALSE))
+  visualize_folders:
+    command: list(
+      tmp='6_visualize/tmp',
+      log='6_visualize/log')
```
[the remark this answers is not in the document]
Comment (jread-usgs) [reply 6_visualize.yml]: Yes - I think using the `placeholder` pattern is best 👍

File: 6_visualize/src/create_gif_tasks.R [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      psprintf(
+        "create_storm_frame(",
+        "png_file=target_name",
+        "config='%s',"=cfg,
+        "'%s',"=view_ind,
+        "'%s',"=basemap_ind,
+        "'%s'"=storm_line_ind,
+        "5_vizprep/out/storm_point_%s.rds.ind,"=cur_task$date_hour,
+        "5_vizprep/out/precip_raster_%s.rds.ind,"=cur_task$date_hour,
+        "5_vizprep/out/streamdata_%s.rds.ind,"=cur_task$date_hour,
```
Comment (aappling-usgs) [inline 6_visualize/src/create_gif_tasks.R]: i think with jordan's pattern these date_hour bits would get wrapped in `[` and `]`?

File: 6_visualize.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jzwart) [reply 6_visualize.yml]: OK I'll do that

**Comment by jzwart** [comment]
yup merge now would be good

File: 6_visualize/src/create_gif_tasks.R (hunk 3 above)
Comment (jzwart) [reply 6_visualize/src/create_gif_tasks.R]: yes will update with new format

File: 6_visualize/src/create_gif_tasks.R (hunk 1 above)
Comment (jzwart) [reply 6_visualize/src/create_gif_tasks.R]: yeah! it's very handy for the task plan creation and I always copy / paste it as is anyways

Comment (jzwart) [reply 6_visualize/src/create_gif_tasks.R]: I can create an issue in scipiper

**Pull request merged**